internal/handlers/user_response_handler.go

```diff
@@ -0,0 +1,437 @@
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	w.WriteHeader(http.StatusCreated)
+	jsonResponse, err := getJSONResponse("token", authToken)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	utils.LogErr(w.Write(jsonResponse))
+}
+
+func getJSONResponse(key string, value string) ([]byte, error) {
+	resp := make(map[string]string)
+	resp[key] = value
+	jsonResp, err := json.Marshal(resp)
+	if err != nil {
+		return jsonResp, fmt.Errorf("error happened in JSON marshal: %w", err)
+	}
+
+	return jsonResp, nil
+}
+
+func getUserPermissions(httpRequest *UserRequest, aRep interfaces.UserActionRepository) ([]models.UserPermission, error) {
+	actions, err := aRep.FindALL()
+	if err != nil {
+		return nil, fmt.Errorf("can't load actions from db: %w", err)
+	}
+
+	var permissions []models.UserPermission
+	for _, product := range httpRequest.Products {
+		for _, permission := range product.Permissions {
+			action, ok := actions[permission]
+			if !ok {
+				return nil, fmt.Errorf("can't find permission: %s", permission)
+			}
+
+			var userPermission models.UserPermission
+			userPermission.ActionID = action.ID
+			userPermission.ProductType = product.Type
+			permissions = append(permissions, userPermission)
+		}
+	}
+
+	return permissions, nil
+}
+
+func getAuthToken() (string, error) {
+	uuidWithHyphen, err := uuid.NewRandom()
+	if err != nil {
+		return "", nil
+	}
+	uuid := strings.Replace(uuidWithHyphen.String(), "-", "", -1)
+
+	return uuid, nil
+}
+
+type ProductRequest struct {
+	Type        string   `validate:"required,max=50"`
+	Permissions []string `validate:"required,oneof='create' 'update' 'delete' 'get'"`
+}
+
+type UserRequest struct {
+	UserId            string `validate:"required,max=64"`
+	Role              string `validate:"required,oneof='service' 'device'"`
+	Title             string `validate:"required,max=100"`
+	Description       string `validate:"max=256"`
+	Products          []ProductRequest
+	IsRegenerateToken bool
+}
+
+func getRequest(r *http.Request) (*UserRequest, error) {
+	var request UserRequest
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&request)
+	if err != nil {
+		return nil, fmt.Errorf("i can't decode json request: %w", err)
+	}
+
+	return &request, nil
+}
+
+func GetUpdatedUserResponse(r *http.Request, w http.ResponseWriter, param UserRouteParameters) {
+	httpRequest, err := getRequest(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	validate := validator.New()
+	err = validate.Struct(httpRequest)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	model, err := param.UserRepository.FindByCode(httpRequest.UserId)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if model == nil {
+		http.Error(
+			w,
+			fmt.Errorf("this user isn't exist, userId=%s", httpRequest.UserId).Error(),
```

Comment:
можно же просто 
```suggestion
			fmt.Sprintf("this user isn't exist, userId=%s", httpRequest.UserId).Error(),
```
так как ты саму ошибку не используешь нигде, а все равно приводишь error к строке

Reply:
аналогично ко всем похожим местам